### PR TITLE
[Bug 1812301] Publish only string typed labels

### PR DESCRIPTION
**Pull request by scholtzan**
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1812301 and should unblock artifact deploys

### CODE REVIEW

**Review by akkomar — APPROVED** [review]
LGTM, but wouldn't it make sense to document that non-string labels are ignored? An inline comment might save future us some time figuring out what's going on.

File: bigquery_etl/metadata/publish_metadata.py [hunk 1]
```diff
@@ -41,7 +41,11 @@ def publish_metadata(client, dataset, table, metadata):
         if metadata.description is not None:
             table.description = metadata.description
 
-        table.labels = metadata.labels
+        table.labels = {
+            key: value
+            for key, value in metadata.labels.items()
+            if isinstance(value, str)
```
Comment (kik-kik) [inline bigquery_etl/metadata/publish_metadata.py:47]: Is there a specific reason to filter stuff out rather than convert to a string?

Comment (kik-kik) [reply bigquery_etl/metadata/publish_metadata.py:47]: Or at the very least should we log that something has been filtered out?

Comment (scholtzan) [reply bigquery_etl/metadata/publish_metadata.py:47]: there are too many restrictions on what characters are allowed in labels (doesn't allow `,`, for example). I don't think it's worth to implement any more complex logic just to get these custom labels which aren't likely be looked at anyway

**Pull request merged**